executor/show_placement.go

```diff
@@ -128,23 +132,117 @@ func (e *ShowExec) fetchShowPlacementLabels(ctx context.Context) error {
 	return nil
 }
 
-func (e *ShowExec) fetchShowPlacement(_ context.Context) error {
-	err := e.fetchAllPlacementPolicies()
+func (e *ShowExec) fetchShowPlacementForTable(_ context.Context) (err error) {
+	tbl, err := e.getTable()
+	if err != nil {
+		return err
+	}
+
+	tblInfo := tbl.Meta()
+	placement, err := e.getTablePlacement(tblInfo)
 	if err != nil {
 		return err
 	}
 
+	if placement != nil {
+		ident := ast.Ident{Schema: e.Table.DBInfo.Name, Name: tblInfo.Name}
+		e.appendRow([]interface{}{"Table " + ident.String(), placement.String()})
+	}
+
 	return nil
 }
 
+func (e *ShowExec) fetchShowPlacement(_ context.Context) error {
+	if err := e.fetchAllPlacementPolicies(); err != nil {
+		return err
+	}
+
+	return e.fetchAllTablePlacements()
+}
+
 func (e *ShowExec) fetchAllPlacementPolicies() error {
 	policies := e.is.AllPlacementPolicies()
 	sort.Slice(policies, func(i, j int) bool { return policies[i].Name.O < policies[j].Name.O })
 	for _, policy := range policies {
 		name := policy.Name
 		settings := policy.PlacementSettings
-		e.appendRow([]interface{}{"POLICY " + name.String(), settings.String(), "SCHEDULED"})
+		e.appendRow([]interface{}{"POLICY " + name.String(), settings.String()})
+	}
+
+	return nil
+}
+
+func (e *ShowExec) fetchAllTablePlacements() error {
+	checker := privilege.GetPrivilegeManager(e.ctx)
+	activeRoles := e.ctx.GetSessionVars().ActiveRoles
+
+	dbs := e.is.AllSchemas()
+	sort.Slice(dbs, func(i, j int) bool { return dbs[i].Name.O < dbs[j].Name.O })
+
+	for _, dbInfo := range dbs {
+		tableRowSets := make([]struct {
+			name string
+			rows [][]interface{}
+		}, 0)
+
+		for _, tbl := range e.is.SchemaTables(dbInfo.Name) {
+			tblInfo := tbl.Meta()
+			if checker != nil && !checker.RequestVerification(activeRoles, dbInfo.Name.O, tblInfo.Name.O, "", mysql.AllPrivMask) {
+				continue
+			}
+
+			var rows [][]interface{}
+			ident := ast.Ident{Schema: dbInfo.Name, Name: tblInfo.Name}
+			placement, err := e.getTablePlacement(tblInfo)
+			if err != nil {
+				return err
+			}
+
+			if placement != nil {
+				rows = append(rows, []interface{}{"Table " + ident.String(), placement.String()})
+			}
+
+			// TODO: Add partition placement rules
+
+			if len(rows) > 0 {
+				tableRowSets = append(tableRowSets, struct {
+					name string
+					rows [][]interface{}
+				}{
+					name: tblInfo.Name.String(),
+					rows: rows,
+				})
+			}
+		}
+
+		sort.Slice(tableRowSets, func(i, j int) bool { return tableRowSets[i].name < tableRowSets[j].name })
```

Comment:
Do we need to ensure the order? If it is for tests, we have `Rows.Sort()` function.

Reply:
It is not defined in the document. But I think it's better to sort it just like what `SHOW DATABASES` and `SHOW TABLES` do.